File: backend/app/utils/social_links.py
def normalize_telegram_link(value: str | None) -> str | None:
    if value is None:
        return None
    raw = value.strip()
    if not raw:
        return None
    if raw.startswith("http://") or raw.startswith("https://"):
        return raw
    if raw.startswith("t.me/"):
        return f"https://{raw}"
    username = raw.lstrip("@")
    if not username:
        return None
    return f"https://t.me/{username}"


def normalize_vk_link(value: str | None) -> str | None:
    if value is None:
        return None
    raw = value.strip()
    if not raw:
        return None
    if raw.startswith("http://") or raw.startswith("https://"):
        return raw
    if raw.startswith("vk.com/") or raw.startswith("m.vk.com/"):
        return f"https://{raw}"
    path = raw.lstrip("@")
    if not path:
        return None
    return f"https://vk.com/{path}"

File: frontend/src/lib/socialLinks.ts
export function normalizeTelegramUrl(value: string | null | undefined): string | null {
  if (!value?.trim()) {return null;}
  const raw = value.trim();
  if (raw.startsWith("http://") || raw.startsWith("https://")) {return raw;}
  if (raw.startsWith("t.me/")) {return `https://${raw}`;}
  const username = raw.replace(/^@/, "");
  return username ? `https://t.me/${username}` : null;
}

export function normalizeVkUrl(value: string | null | undefined): string | null {
  if (!value?.trim()) {return null;}
  const raw = value.trim();
  if (raw.startsWith("http://") || raw.startsWith("https://")) {return raw;}
  if (raw.startsWith("vk.com/") || raw.startsWith("m.vk.com/")) {return `https://${raw}`;}
  const path = raw.replace(/^@/, "");
  return path ? `https://vk.com/${path}` : null;
}
